OSSMDOC-147: Add troubleshooting page. (#29410)
openshift/openshift-docs: service_mesh/v2x/ossm-troubleshooting-istio.adoc

* xref:../../support/troubleshooting/troubleshooting-operator-issues.adoc[OpenShift Operator troubleshooting section].

* xref:../../support/troubleshooting/troubleshooting-installations.adoc[OpenShift installation troubleshooting section]

// The following include statements pull in the module files that comprise the assembly.
include::modules/ossm-troubleshooting-operators.adoc[leveloffset=+1]

include::modules/ossm-troubleshooting-smcp.adoc[leveloffset=+1]

== Troubleshooting the data plane

{ProductName} relies on a proxy sidecar within the application’s pod to provide Service Mesh capabilities to the application.

include::modules/ossm-troubleshooting-injection.adoc[leveloffset=+2]

For more information about sidecar injection, see xref:../../prepare-to-deploy-applications-ossm.adoc#ossm-automatic-sidecar-injection_deploying-applications-ossm[Enabling automatic injection]

include::modules/ossm-troubleshooting-proxy.adoc[leveloffset=2]

For more information about troubleshooting pod issues, see the xref:../../support/troubleshooting/investigating-pod-issues.adoc[OpenShift pod troubleshooting section]

> == Next steps

Author: I added a couple of lines so we don't have an empty heading, and in the process realized that I hadn't moved all the support content out of the Release Notes.  Also added a couple of Support topics from the OpenShift support assembly, so this 
how to get support" section grew a little bit with my last commit.

> == Next steps

Reviewer: The Next steps heading and the line under it are unnecessary here. They should both be removed.

[id="ossm-troubleshooting"]
= Troubleshooting your service mesh
include::modules/ossm-document-attributes.adoc[]
:context: troubleshooting-ossm

toc::[]

This section describes how to identify and resolve common problems in {ProductName}. Use the following sections to help troubleshoot and debug problems when deploying {ProductName} on {product-title}.

In addition to the information in this section, be sure to review the following topics:

* xref:../../support/troubleshooting/troubleshooting-operator-issues.adoc[OpenShift Operator troubleshooting section].

* xref:../../support/troubleshooting/troubleshooting-installations.adoc[OpenShift installation troubleshooting section]

// The following include statements pull in the module files that comprise the assembly.

> == Troubleshooting your installation

Reviewer: I don't think this H2 is necessary. But if we were to keep it, I think it should have a sentence between it and the module. The next thing after a heading should be body text, not another heading. The same goes for `== Troubleshooting the control plane` and `== Troubleshooting the data plane`.

[id="ossm-troubleshooting"]
= Troubleshooting your service mesh
include::modules/ossm-document-attributes.adoc[]
:context: troubleshooting-ossm

toc::[]

This section describes how to identify and resolve common problems in {ProductName}. Use the following sections to help troubleshoot and debug problems when deploying {ProductName} on {product-title}.

// The following include statements pull in the module files that comprise the assembly.

include::modules/ossm-understanding-versions.adoc[leveloffset=+1]

== Troubleshooting Operator installation

In addition to the information in this section, be sure to review the following topics:

> * xref:../../operators/understanding/olm-what-operators-are.adoc[What are Operators?].

Reviewer 2: ```suggestion
* xref:../../operators/understanding/olm-what-operators-are.adoc[What are Operators?]
```

Reviewer 2: https://deploy-preview-29410--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-troubleshooting-istio.html#troubleshooting-operator-installation

* xref:../../operators/understanding/olm/olm-understanding-olm.adoc[Operator Lifecycle Management concepts].

* xref:../../support/troubleshooting/troubleshooting-operator-issues.adoc[OpenShift Operator troubleshooting section].

> * xref:../../support/troubleshooting/troubleshooting-installations.adoc[OpenShift installation troubleshooting section]

Reviewer 2: Either remove all `.` or add one for this last point? Doesn't seem like they should be required for links

include::modules/ossm-validating-operators.adoc[leveloffset=+2]

include::modules/ossm-troubleshooting-operators.adoc[leveloffset=+2]

== Troubleshooting the control plane

> The Service Mesh _control plane_ is composed of *Istiod* which consolidates several previous control plane components (Citadel, Galley, Pilot) into a single binary. Deploying the `ServiceMeshControlPlane` also creates the other components that make up {ProductName} as described in the xref:../../service_mesh/v2x/ossm-architecture.adoc[architecture] topic.

Reviewer 2: ```suggestion
The Service Mesh _control plane_ is composed of Istiod, which consolidates several previous control plane components (Citadel, Galley, and Pilot) into a single binary. Deploying the Service Mesh control plane also creates the other components that make up {ProductName} as described in the xref:../../service_mesh/v2x/ossm-architecture.adoc[architecture] topic.
```

Reviewer 2: Link to this exact heading `openshift-enterprise/latest/service_mesh/v2x/ossm-architecture.html#ossm-architecture_ossm-architecture` instead of the general _Understanding Red Hat OpenShift Service Mesh_ topic

include::modules/ossm-validating-smcp.adoc[leveloffset=+2]

include::modules/ossm-accessing-kiali.adoc[leveloffset=+3]

include::modules/ossm-accessing-jaeger.adoc[leveloffset=+3]

include::modules/ossm-troubleshooting-smcp.adoc[leveloffset=+2]

== Troubleshooting the data plane

The _data plane_ is a set of intelligent proxies that intercept and control all inbound and outbound network communications between services in the service mesh.

{ProductName} relies on a proxy sidecar within the application’s pod to provide service mesh capabilities to the application.

include::modules/ossm-troubleshooting-injection.adoc[leveloffset=+2]

> For more information about sidecar injection, see xref:../../prepare-to-deploy-applications-ossm.adoc#ossm-automatic-sidecar-injection_deploying-applications-ossm[Enabling automatic injection]

Reviewer 2: This link seems broken in the preview?
https://deploy-preview-29410--osdocs.netlify.app/openshift-enterprise/latest/prepare-to-deploy-applications-ossm.html#ossm-automatic-sidecar-injection_deploying-applications-ossm

Author: Fixed

include::modules/ossm-troubleshooting-proxy.adoc[leveloffset=2]

> For more information about troubleshooting pod issues, see xref:../../support/troubleshooting/investigating-pod-issues.adoc[OpenShift pod troubleshooting]

Reviewer 2: ```suggestion
For more information about troubleshooting pod issues, see xref:../../support/troubleshooting/investigating-pod-issues.adoc[Investigating pod issues].
```
I'd use the actual names of the pages so the link doesn't say something different than the heading it goes to

include::modules/ossm-document-attributes.adoc[]
:context: troubleshooting-ossm

toc::[]

This section describes how to identify and resolve common problems in {ProductName}. Use the following sections to help troubleshoot and debug problems when deploying {ProductName} on {product-title}.

// The following include statements pull in the module files that comprise the assembly.

include::modules/ossm-understanding-versions.adoc[leveloffset=+1]

== Troubleshooting Operator installation

In addition to the information in this section, be sure to review the following topics:

* xref:../../operators/understanding/olm-what-operators-are.adoc[What are Operators?].

* xref:../../operators/understanding/olm/olm-understanding-olm.adoc[Operator Lifecycle Management concepts].

* xref:../../support/troubleshooting/troubleshooting-operator-issues.adoc[OpenShift Operator troubleshooting section].

* xref:../../support/troubleshooting/troubleshooting-installations.adoc[OpenShift installation troubleshooting section]

include::modules/ossm-validating-operators.adoc[leveloffset=+2]

include::modules/ossm-troubleshooting-operators.adoc[leveloffset=+2]

== Troubleshooting the control plane

> The Istio control plane is composed of *Istiod* which consolidates several previous control plane components (Citadel, Galley, Pilot) into a single binary.   Deploying the `ServiceMeshControlPlane` also creates the other components that make up {ProductName} as described in the xref:../../service_mesh/v2x/ossm-architecture.adoc[architecture] topic.

Reviewer 2: what is Istiod? A resource, object, etc?

> The Istio control plane is composed of *Istiod* which consolidates several previous control plane components (Citadel, Galley, Pilot) into a single binary.   Deploying the `ServiceMeshControlPlane` also creates the other components that make up {ProductName} as described in the xref:../../service_mesh/v2x/ossm-architecture.adoc[architecture] topic.

Author: It's one of the components of Istio.  I had to fire up a cluster to answer this.   Service Mesh creates an istiod service, pod, deployment, AND replica set.  In this case I had it bold because I was talking about it as a component.

include::modules/ossm-validating-smcp.adoc[leveloffset=+2]

include::modules/ossm-accessing-kiali.adoc[leveloffset=+3]

include::modules/ossm-accessing-jaeger.adoc[leveloffset=+3]

include::modules/ossm-troubleshooting-smcp.adoc[leveloffset=+2]

== Troubleshooting the data plane

> The *data plane* is a set of intelligent proxies that intercept and control all inbound and outbound network communication between microservices in the service mesh.

Reviewer 2: ```suggestion
The _data plane_ is a set of intelligent proxies that intercept and control all inbound and outbound network communications between services in the service mesh.
```
I'd maybe just use "services", microservices and services seem to be used interchangeably earlier in the doc, but I'd choose just one term for simplicity.

> {ProductName} relies on a proxy sidecar within the application’s pod to provide Service Mesh capabilities to the application.

Reviewer 2: again, consistent capitalization on Service Mesh - if referring to the product here, maybe use an attribute?

include::modules/ossm-troubleshooting-injection.adoc[leveloffset=+2]

For more information about sidecar injection, see xref:../../prepare-to-deploy-applications-ossm.adoc#ossm-automatic-sidecar-injection_deploying-applications-ossm[Enabling automatic injection]

include::modules/ossm-troubleshooting-proxy.adoc[leveloffset=2]

> For more information about troubleshooting pod issues, see the xref:../../support/troubleshooting/investigating-pod-issues.adoc[OpenShift pod troubleshooting section]

Reviewer 2: ```suggestion
For more information about troubleshooting pod issues, see xref:../../support/troubleshooting/investigating-pod-issues.adoc[OpenShift pod troubleshooting].
```

include::modules/support.adoc[leveloffset=+1]

include::modules/support-knowledgebase-about.adoc[leveloffset=+2]

include::modules/support-knowledgebase-search.adoc[leveloffset=+2]

include::modules/about-must-gather.adoc[leveloffset=+2]

.Prerequisites

* Access to the cluster as a user with the `cluster-admin` role.

* The {product-title} CLI (`oc`) installed.

include::modules/ossm-about-collecting-ossm-data.adoc[leveloffset=+2]

For prompt support, supply diagnostic information for both {product-title} and {ProductName}.

> include::modules/support-submitting-a-case.adoc[leveloffset=+2]

Reviewer 2: This seems like an odd place to have prereqs - maybe make this a == heading instead if the prereqs apply for a few modules?

Reviewer 2: FYI: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies
